packages/grid/x-data-grid/src/tests/rows.DataGrid.test.tsx

```diff
@@ -1111,6 +1111,14 @@ describe('<DataGrid /> - Rows', () => {
       act(() => apiRef.current.updateRows([{ id: 5, brand: 'Atari' }]));
       expect(getColumnValues(0)).to.deep.equal(['Apple', 'Atari']);
     });
+
+    it('should throw a console error if autoPageSize is used with autoHeight', () => {
+      expect(() => {
+        render(<TestCase autoPageSize autoHeight />);
+      }).toErrorDev(
+        'MUI: The `autoPageSize` prop will conflict with `autoHeight` prop when both of them are enabled.',
+      );
```

Comment:
On the warning message, I would recommend to follow https://www.notion.so/mui-org/Technical-writing-guidance-7e55b517ac2e489a9ddb6d0f6dd765de?pvs=4#85219822c3194b6f8a49ee08ea82b90a.

- Explain what happened, e.g. you provided autoPageSize={true} and autoHeight={true}
- Explain why it's wrong, e.g. these two props can't function together.
- Provide a way out, e.g. please remove one of these two props.

Reply:
is it fine, if I update the error message with the type change PR?